<?xml version="1.0" encoding="UTF-8" ?>
<launch>
  <!-- still no support for other maps!!! -->
  <!-- <arg name="mapname" default="icelab_black" /> -->
  <arg name="planner_type" default="OnlineTaskPlanner"/>
  <arg name="agents_type" default="OnlineAgent"/>
  <arg name="robots_capacity" default="3"/>
  <arg name="gen_type" default="file"/>
  <arg name="mapname" default="icelab_room" />
  <arg name="missions_file" default="1.txt"/>
  <arg name="debug_mode" default="false"/>
  <arg name="gdb_file" default=""/>
  <arg name="robot_order" default="0" />
  <arg name="robot_name" default="rbkairos" />
  <arg name="agent_name" default="patrol_rbkairos" />
  <arg name="interactive_mode" default="false" />
  <arg name="map_frame" default="robot_map" />

  <!-- for now included in tmux script -->
  <!-- <include file="$(find rbkairos_sim_bringup)/launch/rbkairos_complete.launch">
    <arg name="launch_rviz" value="false" />
    <arg name="gazebo_world" value="$(find rbkairos_gazebo)/worlds/icelab.world" />
    <arg name="x_init_pose_robot_a" value="3.09" />
    <arg name="y_init_pose_robot_a" value="1.53" />
  </include> -->

  <include file="$(find logistic_sim)/launch/task_planner.launch">
    <arg name="mapname" value="$(arg mapname)"/>
    <arg name="planner_type" value="$(arg planner_type)"/>
    <arg name="agents_type" value="$(arg agents_type)"/>
    <arg name="agents_number" value="1"/>
    <arg name="robots_capacity" value="$(arg robots_capacity)"/>
    <arg name="gen_type" value="$(arg gen_type)"/>
    <arg name="missions_file" value="$(arg missions_file)"/>
    <arg name="debug_mode" value="$(arg debug_mode)"/>
    <arg name="gdb_file" value="$(arg gdb_file)"/>
  </include>

  <include file="$(find logistic_sim)/launch/agent.launch">
    <arg name="mapname" value="$(arg mapname)" />
    <arg name="agents_type" value="$(arg agents_type)" />
    <arg name="agents_number" value="1" />
    <arg name="robots_capacity" value="$(arg robots_capacity)" />
    <arg name="debug_mode" value="$(arg debug_mode)" />
    <arg name="gdb_file" value="$(arg gdb_file)" />
    <arg name="robot_order" value="$(arg robot_order)" />
    <arg name="robot_name" value="$(arg robot_name)" />
    <arg name="agent_name" value="$(arg agent_name)" />
    <!-- <arg name="map_frame" value="rbkairos_map" /> -->
    <arg name="interactive_mode" value="$(arg interactive_mode)" />
    <arg name="map_frame" value="$(arg map_frame)" />
  </include>
</launch>
